pkg/host-disk/host-disk.go

```diff
@@ -242,27 +251,37 @@ func shouldMountHostDisk(hostDisk *v1.HostDisk) bool {
 }
 
 func (hdc *DiskImgCreator) mountHostDiskAndSetOwnership(vmi *v1.VirtualMachineInstance, volumeName string, hostDisk *v1.HostDisk) error {
-	diskPath := GetMountedHostDiskPathFromHandler(unsafepath.UnsafeAbsolute(hdc.mountRoot.Raw()), volumeName, hostDisk.Path)
-	diskDir := GetMountedHostDiskDirFromHandler(unsafepath.UnsafeAbsolute(hdc.mountRoot.Raw()), volumeName)
-	fileExists, err := ephemeraldiskutils.FileExists(diskPath)
+	diskDir, err := hdc.mountRoot.AppendAndResolveWithRelativeRoot(GetMountedHostDiskDir(volumeName))
 	if err != nil {
-		return err
+		return fmt.Errorf("Resolve diskdir : %v", err)
 	}
-	if !fileExists {
-		if err = hdc.handleRequestedSizeAndCreateSparseRaw(vmi, diskDir, diskPath, hostDisk); err != nil {
+
+	diskPath, err := safepath.JoinNoFollow(diskDir, filepath.Base(hostDisk.Path))
+	fileNotExists := errors.Is(err, unix.ENOENT)
```

Comment:
are you sure you need to go that low level with ENOENT? can't use `errors.Is(err, os.ErrNotExist)`?

Reply:
The `ErrNotExist`is not returned because we use `unix.Openat` is used